app/services/action_builder.rb

```diff
@@ -20,6 +20,15 @@ def build_action(extra_attrs = {})
   def previous_action
```

Comment:
Hey @paulaferris! thanks for the feedback. They are all very valid points, we'll try to update the code to reflect them soon. Thanks!